pkg/deploy/controller/deployerpod/controller.go

```diff
@@ -84,6 +84,7 @@ func (c *DeployerPodController) Handle(pod *kapi.Pod) error {
 	case kapi.PodSucceeded:
 		// Detect failure based on the container state
 		nextStatus = deployapi.DeploymentStatusComplete
+		delete(deployment.Annotations, deployapi.DesiredReplicasAnnotation)
```

Comment:
`nextStatus` is assigned to complete optimistically; the `for` loop immediately following might set it to failed, in which case the annotation should be preserved.

Let's add test coverage for this state.